x-pack/plugins/apm/public/hooks/use_apm_api_transaction.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { apm, Span } from '@elastic/apm-rum';
+import { useEffect, useMemo } from 'react';
+import { FETCH_STATUS } from './use_fetcher';
+import { useApmParams } from './use_apm_params';
+import { getParsedDate } from '../context/url_params_context/helpers';
+
+export function useFetcherSpan(endpoint: string, status: FETCH_STATUS) {
+  return useMemo(() => ({ endpoint, status }), [endpoint, status]);
+}
+
+export function useFetcherTransaction(
```

Comment:
have you considered implementing this in `callApmApi`? I can imagine that we can more easily hook into certain phases of the request, but I haven't fully thought it through.

Reply:
I did look into it. The problem is that if you don't specific which requests you are interested in tracking, you'll wait for every request on the page to finish before ending the transaction. Or worse: the transaction will automatically end before all requests have started.

Reply:
but you can still specify which requests you are tracking, but as a prop to callApmApi, no?

Reply:
How would you handle starting and ending the transaction if done like this?